content/hi/docs/setup/production-environment/container-runtimes.md

```diff
@@ -0,0 +1,407 @@
+---
+title: कंटेनर रनटाइम
+content_type: concept
+weight: 20
+---
+<!-- overview -->
+
+आपको क्लस्टर में प्रत्येक नोड में एक {{< glossary_tooltip text="container रनटाइम" term_id="container-runtime" >}} 
+इंस्टॉल करना होगा ताकि पॉड वहां चल सकें। 
+यह पृष्ठ बताता है कि क्या शामिल है और 
+नोड्स की स्थापना के लिए संबंधित कार्यों का वर्णन करता है।
+
+<!-- body -->
+
+यह पृष्ठ लिनक्स पर कुबेरनेट्स के साथ कई सामान्य कंटेनर रनटाइम का उपयोग करने के विवरण सूचीबद्ध करता है: 
+
+- [containerd](#containerd)
+- [CRI-O](#cri-o)
+- [Docker](#docker)
+
+{{< note >}}
+अन्य ऑपरेटिंग सिस्टम के लिए, अपने प्लेटफ़ॉर्म के लिए अनुसार विशिष्ट प्रलेखन देखें।
+{{< /note >}}
+
+## Cgroup ड्राइवर 
+
+प्रक्रियाओं के लिए आवंटित संसाधनों को सीमित करने के लिए कण्ट्रोल ग्रुप का उपयोग किया जाता है।
+
+जब Linux वातावरण के लिए init सिस्टम, [systemd]
+(https://www.freedesktop.org/wiki/Software/systemd/)
+को चुना जाता है, तब init प्रक्रिया root control group(`cgroup`) उत्पन्न करती है और उपभोग करती है 
+तथा cgroup मैनेजर  की तरह काम करता है।
+
+
+Systemd और cgroups एकीकृत हैं और प्रत्येक systemd यूनिट को एक cgroup आवंटित होता है। अपने कन्टैनर रनटाइम और kubelet को `cgroupfs` प्रयोग करने के लिए कॉन्फ़िगर करना संभव है। systemd के साथ `cgroupfs` प्रयोग करने के कारण दो अलग cgroup मैनेजर होंगे।
+
+एक एकल cgroup प्रबंधक इस दृष्टिकोण को सरल बनाता है कि कौन से संसाधन आवंटित किए जा रहे हैं और डिफ़ॉल्ट रूप से उपलब्ध और उपयोग में आने वाले संसाधनों के बारे में अधिक सुसंगत दृश्य होगा।
+जब एक सिस्टम पर दो cgroup मैनेजर होते हैं, तो आपको उन रिसोर्सेज के दो व्यू मिलते हैं। क्षेत्र में, लोगों ने ऐसे मामलों की सूचना दी है जहां नोड्स जो kubelet और डॉकर के लिए `cgroupfs` का उपयोग करने के लिए कॉन्फ़िगर किए गए हैं, लेकिन बाकी प्रक्रियाओं के लिए `systemd` संसाधन दबाव के कारण अस्थिर हो जाते हैं। 
+
+सेटिंग्स को इस तरह बदलना कि आपका कंटेनर रनटाइम और kubelet `systemd` का उपयोग करें क्योंकि cgroup ड्राइवर सिस्टम को स्थिर करता है। डॉकर के लिए इसे कॉन्फ़िगर करने के लिए, `native.cgroupdriver=systemd` सेट करें। 
+
+{{<caution >}}
+क्लस्टर में शामिल होने वाले नोड के cgroup ड्राइवर को बदलना एक संवेदनशील ऑपरेशन है। यदि kubelet ने एक सीग्रुप ड्राइवर के शब्दार्थ का उपयोग करके पॉड्स बनाए हैं, तो कंटेनर रनटाइम को दूसरे सीग्रुप ड्राइवर में बदलने से मौजूदा पॉड्स के पॉड सैंडबॉक्स को फिर से बनाते समय त्रुटियां हो सकती हैं। kubelet को पुनरारंभ करने से ऐसी त्रुटियों का समाधान नहीं हो सकता है। 
+यदि आपके पास स्वचालन है जो इसे व्यवहार्य बनाता है, तो अद्यतन किए गए कॉन्फ़िगरेशन का उपयोग करके नोड को दूसरे के साथ बदलें, या स्वचालन का उपयोग करके इसे पुनर्स्थापित करें।
+
+{{< /caution >}}
+
+## Cgroup v2
+
+Cgroup v2, cgroup Linux API का अगला संस्करण है।cgroup v1 से अलग, प्रत्येक कंट्रोलर के लिए एक अलग अनुक्रम के बजाय एक एकल पदानुक्रम है। 
```

Comment:
Does this make sense?
```suggestion
## cgroup संस्करण 2 {#cgroup-v2}
```

Reply:
Yep, it does, It's the same as saying "cgroup version 2" in English but we can't do acronyms like "v2" in Hindi because acronyms work differently.